components/common/layout/PageLayout.tsx

```diff
@@ -31,6 +31,31 @@ const PageLayout: FC<PageLayoutProps> = ({
   blogPage = false,
   homePage = false,
 }): ReactElement => {
+  const [showButton, setShowButton] = useState(false);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      if (window.scrollY > 300) {
+        setShowButton(true);
+      } else {
+        setShowButton(false);
+      }
+    };
+
+    window.addEventListener('scroll', handleScroll);
+
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  const scrollToTop = () => {
```

Comment:
I know you're closing this, but all you need instead of a button and useEffect etc. is a link. There is a native `#top` anchor.

<a href="#top">Back to top</a> and you can enable smooth scrolling to make it work SMOOOOTHH.